From 22713c6c6c7ed238436e9fcb028172af9731f62a Mon Sep 17 00:00:00 2001
From: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
Date: Wed, 19 Jun 2013 14:04:00 +0200
Subject: [PATCH 1028/1083] [HACK] drm/rcar-du: Add default modes to VGA
 connector

According to the DRM developers drivers must not manually add default
modes to VGA connectors in an unknown connection state. Until a proper
solution is found (possibly by handling default modes in the DRM core,
as for connectors in the connected state), this hack is a convenience
that avoids forcing userspace to configure the modes manually.

Signed-off-by: Laurent Pinchart <laurent.pinchart+renesas@ideasonboard.com>
(cherry picked from commit 362dbf967c2f03b7c2e714701be250cc873e77db)

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit aa4b1eba72d3bd491c1e32d720b99b92ae098733)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_vgacon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_vgacon.c b/drivers/gpu/drm/rcar-du/rcar_du_vgacon.c
index 36105db..719889d 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_vgacon.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_vgacon.c
@@ -22,7 +22,7 @@
 
 static int rcar_du_vga_connector_get_modes(struct drm_connector *connector)
 {
-	return 0;
+	return drm_add_modes_noedid(connector, 1280, 768);
 }
 
 static int rcar_du_vga_connector_mode_valid(struct drm_connector *connector,
-- 
1.8.3.2

